VultisigApp/VultisigApp/Services/Thorchain/ThorchainService.swift

```diff
@@ -15,16 +15,88 @@ class ThorchainService: ThorchainSwapProvider {
     
     private init() {}
     
-    func fetchBalances(_ address: String) async throws -> [CosmosBalance] {
-        guard let url = URL(string: Endpoint.fetchAccountBalanceThorchainNineRealms(address: address)) else        {
+    func fetchBalances(coin: Coin) async throws -> [CosmosBalance] {
```

Comment:
`Coin` is backed by swift data , so prefer we don't pass it around , as you might trigger unnecessary database access
Can you please just pass the address? or whatever parameter you need